packages/flutter/lib/src/widgets/focus_manager.dart

```diff
@@ -1873,6 +1874,16 @@ class FocusManager with DiagnosticableTreeMixin, ChangeNotifier {
     assert(_focusDebug(() => 'Notified ${_dirtyNodes.length} dirty nodes:', () => _dirtyNodes));
     _dirtyNodes.clear();
     if (previousFocus != _primaryFocus) {
+      if (_primaryFocus != null && (_primaryFocus!.context?.mounted ?? false)) {
+        final int? viewId = View.maybeOf(_primaryFocus!.context!)?.viewId;
+        if (viewId != null) {
```

Comment:
If viewId is null then we need to let the engine know that the previously focused view (that's what the latest focused view bit is for) lost focus, hence it should be blurred.  See https://github.com/flutter/engine/blob/main/lib/web_ui/lib/src/engine/platform_dispatcher/view_focus_binding.dart#L57

Reply:
If the view id is null, then this widget isn't in a view anymore, and shouldn't be receiving focus, much less notifying the engine of its focus change. This state (the view being null) should probably never happen, since anything that is no longer part of a view should be disposed before it gets here, and thus never have a chance to become the primary focus anyhow. The only node in the focus tree that this might legitimately happen to is the root node, which doesn't have a context and so would also not report.

We could keep track of what the view used to be the last time any node reported a focus change, but then why not just use the one kept in the engine, since the engine is controlling which view is focused anyhow?